### PR TITLE
Use a dict for looking up table subitem readers

### PR DESCRIPTION
**Pull request by ch3pjw**
This seems like a more compact and efficient way to read table subitems.

### CODE REVIEW

**Comment by dzen** [comment]
Nice point :)

File: aioamqp/frame.py [hunk 1]
```diff
@@ -283,53 +283,41 @@ def read_table(self):
...
-            return table_data.read_table()
-        elif value_type == 'V':
+        if value_type == 'V':
             return None
         else:
-            raise ValueError('Unknown value_type {}'.format(value_type))
+            reader_name = self._table_subitem_reader_map.get(value_type)
+            if not reader_name:
+                raise ValueError('Unknown value_type {}'.format(value_type))
+            return getattr(table_data, reader_name)()
```
Comment (dzen) [inline aioamqp/frame.py]: I'm not fand of lonely parenthesis in the code like thoses :(

Comment (ch3pjw) [inline aioamqp/frame.py]: Are you proprosing

```python
attr = getattr(table_data, reader_name)
return attr()
```

Comment (dzen) [inline aioamqp/frame.py]: For instance, yes

**Comment by ch3pjw** [comment]
Is there anything outstanding on this that you want from my end?

**Comment by foolswood** [comment]
@dzen If the parens thing gets resolved is this ready to go?

**Comment by dzen** [comment]
There was nothing really important in there.

cherry-picked and pushed on master with 9e46956.

I really prefer to see a flat history.

**Comment by ch3pjw** [comment]
I can tell you like the flat history, it makes tracking whether my branch is merged rather a lot harder though, as github has no-way of correlating your cheery-pick with this PR or my branch.

**Pull request closed without merging**